compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10533,21 +10531,18 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
       TR::Register *baseReg1Temp = NULL;
       TR::Register *baseReg2Temp = NULL;
 
+
+      // if any of the to be generated CLC instructions will need a large disp or the reference offsets are uneven, generate an LA up front.
```

Comment:
After this we can be sure that both references have equal offsets and it has enough capacity for the clc instruction.